agent/inbound.go

```diff
@@ -374,6 +374,12 @@ func (a *Agent) createApplication(incoming *v1alpha1.Application) (*v1alpha1.App
 		return created, nil
 	}
 
+	if a.mode == types.AgentModeManaged {
+		// When sync status changed to Synced, trigger a cluster cache info update event
+		// so that the principal can be informed of the cache update.
+		go a.addClusterCacheInfoUpdateToQueue("application_created")
```

Comment:
We're probably fine to remove `go` and run it on the same thread (presuming we remove the sleep from `addClusterCacheInfoUpdateToQueue`)

If we DONT remove `go`, then we likely need some other mechanism (lock) to ensure that we don't have multiple go-routine all updates the cluster info at the same time,